packages/signature-controller/src/SignatureController.ts

```diff
@@ -406,6 +426,45 @@ export class SignatureController extends BaseControllerV2<
     }
   }
 
+  /**
+   * Called when MMI needs to set the defer value in the message.
+   *
+   * @param messageId - The id of the Message to update.
+   * @param deferAsSigned - The defer value to add in the message params.
+   */
+  setDeferSetAsSigned(messageId: string, deferAsSigned: any) {
+    const messageManagers = [
+      this.#messageManager,
+      this.#personalMessageManager,
+      this.#typedMessageManager,
+    ];
+
+    for (const manager of messageManagers) {
+      if (this.#trySetDeferSetAsSigned(manager, messageId, deferAsSigned)) {
+        return;
+      }
+    }
+  }
+
+  /**
+   * Called when a MMI wants to cancel a signing message.
+   *
+   * @param messageId - The id of the Message to update.
+   */
+  cancelAbstractMessage(messageId: string) {
+    const messageManagers = [
+      this.#messageManager,
+      this.#personalMessageManager,
+      this.#typedMessageManager,
+    ];
+
+    for (const manager of messageManagers) {
```

Comment:
As we've done this pattern in a few places, would it warrant a `#tryForEachMessageManager` method which attempts a callback on each message manager until it returns `true`?